source/MaterialXGenShader/Util.cpp

```diff
@@ -411,7 +428,54 @@ bool isTransparentSurface(ElementPtr element, const ShaderGenerator& shadergen)
                     const OutputPtr& output = outputs[0];
                     if (TypeDesc::get(output->getType()) == Type::SURFACESHADER)
                     {
-                        return isTransparentShaderGraph(output, shadergen);
+                        StringSet opacityInterfaceNames;
+                        StringSet transmissionInterfaceNames;
+                        bool isTransparent = isTransparentShaderGraph(output, shadergen, opacityInterfaceNames, transmissionInterfaceNames);
+
+                        if (!isTransparent)
+                        {
+                            for (const string& transmissionInterfaceName : transmissionInterfaceNames)
+                            {
+                                // First check the transmission weight
+                                BindInputPtr transmission = shaderRef->getBindInput(transmissionInterfaceName);
+                                if (transmission)
+                                {
+                                    if (!transmission->getOutputString().empty())
+                                    {
+                                        isTransparent = true;
+                                    }
+                                    else
+                                    {
+                                        ValuePtr value = transmission->getValue();
+                                        if (value && !isZero(value->asA<float>()))
+                                        {
+                                            isTransparent = true;
+                                        }
+                                    }
+                                }
+                            }
+                            for (const string& opacityInterfaceName : opacityInterfaceNames)
```

Comment:
Scan the found interfacenames to see if the shaderref instance has used them in bindinputs. If so then check for upstream connection (any implies transparent) or if value indicates transparency.